vic/drivers/classic/src/parse_output_info.c

```diff
@@ -31,79 +31,126 @@
+    if (nstreams > 0) {
+        options.Noutstreams = nstreams;
 
-        *out_data_files = calloc(options.Noutfiles, sizeof(*(*out_data_files)));
-        if (*out_data_files == NULL) {
+        output_streams = calloc(options.Noutstreams, sizeof(*(output_streams)));
+        if (output_streams == NULL) {
+            log_err("Memory allocation error in parse_output_info().");
+        }
+        out_data_files = calloc(options.Noutstreams, sizeof(*(out_data_files)));
+        if (out_data_files == NULL) {
             log_err("Memory allocation error in parse_output_info().");
         }
-        init_output_list(out_data, false, "%.4g", OUT_TYPE_FLOAT, 1);
-
-        // PRT_SNOW_BAND is ignored if options.Noutfiles > 0
-        options.PRT_SNOW_BAND = false;
 
         while (!feof(gp)) {
             if (cmdstr[0] != '#' && cmdstr[0] != '\n' && cmdstr[0] != '\0') {
                 sscanf(cmdstr, "%s", optstr);
 
                 if (strcasecmp("OUTFILE", optstr) == 0) {
-                    outfilenum++;
-                    if (outfilenum >= (short int)options.Noutfiles) {
+                    streamnum++;
+                    if (streamnum >= (short int) options.Noutstreams) {
                         log_err("Found too many output files, was expecting "
-                                "%zu but found %hu", options.Noutfiles,
-                                outfilenum);
+                                "%zu but found %hu", options.Noutstreams,
+                                streamnum);
                     }
-                    sscanf(cmdstr, "%*s %s",
-                           (*out_data_files)[outfilenum].prefix);
+                    sscanf(cmdstr, "%*s %s", out_data_files[streamnum]->prefix);
 
                     // determine how many variable will be in this file before
                     // allocating (GH: 209)
-                    (*out_data_files)[outfilenum].nvars =
-                        count_outfile_nvars(gp);
+                    nvars = count_outfile_nvars(gp);
+
+                    // TODO: parse stream specific agg period info from global param
+                    // nextagg = (unsigned int) ((int) global_param.out_dt / (int) global_param.dt);
+                    nextagg = 1;
+
+                    setup_stream(output_streams[streamnum],
+                                 out_data_files[streamnum],
+                                 nvars, nextagg);
 
-                    (*out_data_files)[outfilenum].varid =
-                        calloc((*out_data_files)[outfilenum].nvars,
-                               sizeof(*((*out_data_files)[outfilenum].varid)));
-                    if ((*out_data_files)[outfilenum].varid == NULL) {
+                    outvarnum = 0;
+                }
+                else if (strcasecmp("OUTPUT_STEPS_PER_DAY", optstr) == 0) {
+                    sscanf(cmdstr, "%*s %zu",
+                           &out_data_files[streamnum]->output_steps_per_day);
```

Comment:
I *think* you can remove the `&` here and it will have the desired effect.  Otherwise you'd have to do `&(*out_data_files[streamnum])->output_steps_per_day` but that seems really cumbersome - hopefully simply removing the `&` is sufficient...  or perhaps you could do `(*out_data_files)[streamnum].output_steps_per_day`...   anyway somehow this needs to be consistent with passing the address of out_data_files to `parse_output_info`